client/trino-client/src/main/java/io/trino/client/OkHttpUtil.java

```diff
@@ -297,6 +305,25 @@ private static KeyStore loadTrustStore(File trustStorePath, Optional<String> tru
         return trustStore;
     }
 
+    private static KeyStore loadSystemKeyStore(Optional<String> keyStoreType)
+            throws IOException, GeneralSecurityException
+    {
+        String osName = Optional.ofNullable(StandardSystemProperty.OS_NAME.value()).orElse("");
+        Optional<String> systemKeyStoreType = keyStoreType;
+        if (!systemKeyStoreType.isPresent()) {
```

Comment:
Do we need to add something for Linux since that is actually our main supported operating system .. neither Windows or MacOS are ..

Reply:
Initially I just added support for Windows. However I wanted to ensure feature parity in the client with the system trust store option so added MacOS too.

I'd be interested in Linux support as well, but as far as I know there isn't a standardised way to do this that is supported out of the box with [JDK Providers](https://docs.oracle.com/en/java/javase/22/security/oracle-providers.html)?

I guess this is probably due to the proliferation of different Linux keystore implementations, some of which are provided by the different desktop environments and some by various auth tooling.